web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -0,0 +1,297 @@
+  const startSshSession = (login: string, serverId: string) => {
+    const url = cfg.getSshConnectRoute({
+      clusterId,
+      serverId,
+      login,
+    });
+
+    openNewTab(url);
+  };
+
+  function handleOnOpen() {
+    return makeNodeOptions(clusterId, node);
+  }
+
+  const handleOnSelect = (e: React.SyntheticEvent, login: string) => {
+    e.preventDefault();
+    return startSshSession(login, node.id);
+  };
+
+  return (
+    <MenuLogin
+      getLoginItems={handleOnOpen}
+      onSelect={handleOnSelect}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'right',
+      }}
+      anchorOrigin={{
+        vertical: 'center',
+        horizontal: 'right',
+      }}
+    />
+  );
+};
+
+const DesktopConnect = ({ desktop }: { desktop: Desktop }) => {
+  const { clusterId } = useStickyClusterId();
+  const startRemoteDesktopSession = (username: string, desktopName: string) => {
+    const url = cfg.getDesktopRoute({
+      clusterId,
+      desktopName,
+      username,
+    });
+
+    openNewTab(url);
+  };
+
+  function handleOnOpen() {
+    return makeDesktopLoginOptions(clusterId, desktop.name, desktop.logins);
+  }
+
+  function handleOnSelect(e: React.SyntheticEvent, login: string) {
+    e.preventDefault();
+    return startRemoteDesktopSession(login, desktop.name);
+  }
+
+  return (
+    <MenuLogin
+      getLoginItems={handleOnOpen}
+      onSelect={handleOnSelect}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'right',
+      }}
+      anchorOrigin={{
+        vertical: 'center',
+        horizontal: 'right',
+      }}
+    />
+  );
+};
+
+const AppLaunch = ({ app }: { app: App }) => {
+  const {
+    launchUrl,
+    awsConsole,
+    awsRoles,
+    fqdn,
+    clusterId,
+    publicAddr,
+    isCloudOrTcpEndpoint,
+    samlApp,
+    samlAppSsoUrl,
+  } = app;
+  let $btn;
+  if (awsConsole) {
+    $btn = (
+      <AwsLaunchButton
+        awsRoles={awsRoles}
+        fqdn={fqdn}
+        clusterId={clusterId}
+        publicAddr={publicAddr}
+      />
+    );
+  } else if (isCloudOrTcpEndpoint) {
+    $btn = (
+      <ButtonBorder
+        disabled
+        width="90px"
+        size="small"
+        title="Cloud or TCP applications cannot be launched by the browser"
+      >
+        Launch
+      </ButtonBorder>
+    );
+  } else if (samlApp) {
+    $btn = (
+      <ButtonBorder
+        as="a"
+        width="90px"
+        size="small"
+        target="_blank"
+        href={samlAppSsoUrl}
+        rel="noreferrer"
+      >
+        Login
+      </ButtonBorder>
+    );
+  } else {
+    $btn = (
+      <ButtonBorder
+        as="a"
+        width="90px"
+        size="small"
+        target="_blank"
+        href={launchUrl}
+        rel="noreferrer"
+      >
+        Launch
+      </ButtonBorder>
+    );
+  }
+
+  return $btn;
+};
+
+function DatabaseConnect({ database }: { database: Database }) {
+  const { name, protocol } = database;
+  const ctx = useTeleport();
```

Comment:
If you do it, you also need to make a context provider wrapper in `ResourceCard.story.tsx`; otherwise, these won't render. :/ Sorry, I didn't think about it previously.

Reply:
No worries, added the context 👍